src/rocm_smi.cc

```diff
@@ -4013,7 +4013,10 @@ rsmi_event_notification_get(int timeout_ms,
            reinterpret_cast<rsmi_evt_notification_data_t *>(&data[*num_elem]);
 
       uint32_t event;
-      while (fscanf(anon_fp, "%x %63s\n", &event,
+      #define __LEN__(X) #X
+      #define LEN(X) __LEN__(X)
+
+      while (fscanf(anon_fp, "%x %" LEN(MAX_EVENT_NOTIFICATION_MSG_SIZE)"[^\n]\n", &event,
```

Comment:
KFD will return message defined as char fifo_in[KFD_SMI_EVENT_MSG_SIZE], written by snprintf, the string will be KFD_SMI_EVENT_MSG_SIZE - 1 plus null termination. But for the safety, in case KFD increase the string length in future, this should be %(MAX_EVENT_NOTIFICATION_MSG_SIZE - 1), as data.message is defined char message[MAX_EVENT_NOTIFICATION_MSG_SIZE], fscanf will need one extra byte to store null termination at end of string.